### ISSUE TITLE
Crash : ConcurrentModificationException

### ISSUE THREAD

**Issue by SuhasTS** [CleverTap/clevertap-android-sdk#132]
We are using clevertap product config in our project. There were many crashes reported related to  `CTProductConfigController.java` class.

**Stack trace** - 
```
Fatal Exception: java.util.ConcurrentModificationException
       at java.util.HashMap$HashIterator.nextNode(HashMap.java:1441)
       at java.util.HashMap$EntryIterator.next(HashMap.java:1475)
       at java.util.HashMap$EntryIterator.next(HashMap.java:1473)
       at java.util.AbstractMap.toString(AbstractMap.java:554)
       at java.lang.String.valueOf(String.java:2924)
       at java.lang.StringBuilder.append(StringBuilder.java:132)
       at com.clevertap.android.sdk.product_config.CTProductConfigController$1.onPostExecute(CTProductConfigController.java:109)
       at com.clevertap.android.sdk.product_config.CTProductConfigController$1.onPostExecute(CTProductConfigController.java:78)
       at com.clevertap.android.sdk.TaskManager$1$1.run(TaskManager.java:74)
       at android.os.Handler.handleCallback(Handler.java:883)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:7807)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1047)
```

**Comment by kumaratulismu** [comment]
Hi @SuhasTS Thanks for reporting this.

Are you calling Product Config fetch/activate from background thread?

Can you please confirm ?

**Comment by SuhasTS** [comment]
We are calling `fetch` on callback of `onInit()`  and `activate` on callback of `onFetched()`. Need to check on what threads these callbacks are fired.

**Comment by kumaratulismu** [comment]
@SuhasTS We have fixed this in our latest release v4.0.3. 
Closing this.
